src/sinks/util/tls.rs

```diff
@@ -55,32 +54,37 @@ enum TlsError {
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 pub struct TlsOptions {
     pub verify_certificate: Option<bool>,
+    pub verify_hostname: Option<bool>,
     pub ca_path: Option<PathBuf>,
     pub crt_path: Option<PathBuf>,
     pub key_path: Option<PathBuf>,
     pub key_pass: Option<String>,
 }
 
-impl TlsOptions {
-    pub fn check_warnings(&self, sink: &str) {
-        if self.verify_certificate == Some(false) {
-            warn!(
-                "`verify_certificate` in {} sink is DISABLED, this may lead to security vulnerabilities", sink
-            );
-        }
-    }
-}
-
 /// Directly usable settings for TLS connectors
+#[derive(Clone, Default)]
 pub struct TlsSettings {
-    verify_certificate: bool,
+    accept_invalid_certificates: bool,
+    accept_invalid_hostnames: bool,
     authority: Option<Certificate>,
-    identity: Option<Identity>,
+    identity: Option<Vec<u8>>, // native_tls::Identity doesn't implement Clone yet
```

Comment:
minor nit: But since this buf represents something maybe it might be nice to throw this under a newtype?

Reply:
Definitely, good call. Done.